src/coreclr/jit/codegenlinear.cpp

```diff
@@ -755,6 +754,16 @@ void CodeGen::genCodeForBBlist()
 
             case BBJ_ALWAYS:
             {
+#if DEBUG
```

Comment:
This is related to the comments about adjusting/unadjusting the IP of the return site by `-1`. That, according to the comments, is to handle a case where location right after a throwing call could be in a different BB with completely unrelated liveness set. 
I do not think it is possible and there is always at least a NOP as such fallthrough might break other things. 
I think we should stop doing that adjustment (it would be a breaking change though). This assert just checks that such fall-through does not happen.